src/main.cpp

```diff
@@ -92,6 +96,8 @@ int main()
             0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0,
             0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0,
         }};
+		
+		plottingMatrix[i] = new matrix{8, 1000, new float[8000]}; 
```

Comment:
you already have an double array plottingMatrix[8][1000]. why do you declare each of of the ith pointer to be of size 8000 shouldn't they be of size 1000 each so 8x1000 in total is 8000??